app/assets/javascripts/controllers/HoldCancelCtrl.js

```diff
@@ -16,6 +16,12 @@ app.controller('HoldCancelCtrl', [ '$scope', '$routeParams', '$http', '$location
         return false;
       }
 
+      if (data.hold.status == "cancelled"){
+        $('#cancel-text').html("This current order of <strong>" + data.teacher_set.title + "</strong> has already beeen cancelled!")
```

Comment:
To check for `null`, change this:
`"This current order of <strong>" + data.teacher_set.title + "</strong> has already beeen cancelled!"`
to this:
`"This order is already cancelled."`

There was a typo in "beeen".

Reply:
Agreed that we need to secure string generation against null.  Another way to do it is to use string interpolation, and check for null or underfined values in there.  There's an example in the holds email generation code.

Reply:
@LucasCioffi @daryanypl Have you seen a case where a teacher set had a title of null?  I checked the db and did not. I think it would be better to add a validation for the teacher_set model preventing any records with the title nil to be inserted. Therefore, avoiding this issue all together.

Reply:
@nnat425 Darya is checking with Courtney about minimum fields.  If the ILS accepts a book created without a title (not sure if possible) and we reject that record from MLN, then the two databases will be out of sync.  Courtney estimates that she might be done with that by the end of today.  Given this, I'd recommend checking for null and we can implement validation based on what Courtney recommends.

Reply:
So it sounds like first, we need to confirm whether the ILS accepts a teacher_set without a title. Second if it does, how are you going to determine the title of the teacher_set? If the ILS does not accept a teacher_set without a title, then the solution I coded will work no problem.

Regardless, I'll check for null here but I believe its redundant code that doesn't need to be here at all.